packages/rich-text/src/get-active-formats.js

```diff
@@ -5,19 +5,26 @@
  *
  * @return {?Object} Active format objects.
  */
-export function getActiveFormats( { formats, start, selectedFormat } ) {
+export function getActiveFormats( { formats, start, end, activeFormats } ) {
 	if ( start === undefined ) {
 		return [];
 	}
 
-	const formatsBefore = formats[ start - 1 ] || [];
-	const formatsAfter = formats[ start ] || [];
+	if ( start === end ) {
+		// For a collapsed caret, it is possible to override the active formats.
+		if ( activeFormats ) {
+			return activeFormats;
+		}
 
-	let source = formatsAfter;
+		const formatsBefore = formats[ start - 1 ] || [];
+		const formatsAfter = formats[ start ] || [];
 
-	if ( formatsBefore.length > formatsAfter.length ) {
-		source = formatsBefore;
+		if ( formatsBefore.length < formatsAfter.length ) {
```

Comment:
By default, we'd like to "select" the lowest amount of formats. E.g. when the user puts the caret at a boundary, it will be outside the boundary by default. Perhaps this could use a comment.

Reply:
> By default, we'd like to "select" the lowest amount of formats. E.g. when the user puts the caret at a boundary, it will be outside the boundary by default. Perhaps this could use a comment.

That makes sense, and I agree could benefit from a comment 👍